projects/hyperdash-angular/src/editor/nested-model/nested-model-editor.component.ts

```diff
@@ -9,9 +9,7 @@ import { EDITOR_API } from '../editor-api-injection-token';
 @Component({
   selector: 'hda-nested-model-editor',
   changeDetection: ChangeDetectionStrategy.OnPush,
-  template: `
-    <hda-model-json-editor [modelJson]="api.value" (modelJsonChange)="api.valueChange($event)"> </hda-model-json-editor>
-  `
+  template: ' To be Added! '
```

Comment:
please don't break the working component - just remove the unused dependency that was causing the cycle

Reply:
Yeah, I need your input. Below is the problematic method. 

```
  private getRendererComponentForEditorData(editorData: NestedEditorData): Type<unknown> {
    switch (editorData.kind) {
      case EditorKind.Leaf:
        return editorData.editor;
      case EditorKind.Unresolved:
      default:
        // Should never be a default
        return NestedModelEditorComponent;
    }
  }
```

I could throw an error for `Unresolved` and `default` case. Would this be better?

```
  private getRendererComponentForEditorData(editorData: NestedEditorData): Type<unknown> {
    switch (editorData.kind) {
      case EditorKind.Leaf:
        return editorData.editor;
      case EditorKind.Unresolved:
      default:
        throw Error('Unsupported code path');
    }
  }
```

Reply:
What if you remove the service dep from https://github.com/hypertrace/hyperdash-angular/blob/3b28405a452b8b56522c67fd5fdf990bb7817a98/projects/hyperdash-angular/src/editor/model-json/model-json-editor.component.ts ? Unless I'm misreading it, it's only using it to assign an unused variable (can't recall the backstory though)

Reply:
that wouldn't help. The dependency is also coming from `'hda-nested-model-editor'` -> `hda-model-json-editor` -> `<hda-model-editor` -> `model-editor.service` ->  `'hda-nested-model-editor'`. 

I would propose that we remove the last dep with this code 
```
case EditorKind.Unresolved: 
default: throw Error('Unsupported code path');
```

Reply:
that's fine, we can come back to it if we resurrect the editor stuff.

Reply:
Instead of throwing error, I am returning undefined and filtering the unknown case.

```
  private getRendererComponentForEditorData(editorData: NestedEditorData): Type<unknown> | undefined {
    switch (editorData.kind) {
      case EditorKind.Leaf:
        return editorData.editor;
      case EditorKind.Unresolved:
      default:
        return undefined;
    }
  }
  
  ```

Reply:
So with this change, can we restore the template here?

Reply:
aah. I thought I reverted it.

Reply:
done